handbook/engineering/developer-insights/code-insights/processes.md

```diff
@@ -16,28 +16,80 @@ Attendees are encouraged to add comments on talking points asynchronously before
+    As a rule of thumb, the sum of estimates for each individual should not exceed **8 days** (out of a 10-work-day iteration) so we have enough buffer, time to reply to asks from other teams, to plan the next iteration's projects, and self-organized slack time.
+
+- During an iteration, teammates **work on their assigned issues for the iteration in the order they are listed** in the ["Current iteration" view](https://github.com/orgs/sourcegraph/projects/200/views/1) of the board. When starting work on a task, the teammate **updates its status column to "In Progress"** to communicate it to their team. This gives a good overview in the ["Current iteraton" view](https://github.com/orgs/sourcegraph/projects/200/views/1), which can also be viewed in [Kanban layout](https://github.com/orgs/sourcegraph/projects/200/views/1?layout=board), on how the iteration is tracking.
+
+- If one or more issues that were planned for an iteration are looking to **not get finished** (which includes testing) in the [current iteration](https://github.com/orgs/sourcegraph/projects/200/views/1) (while maintaining sustainable work practices) the assignee **raises this as soon as possible asynchronously** to the team (including the PM and EM), e.g. on the GitHub issue or Slack. These issues then **become _proposed_ issues for the next iteration** (meaning nothing carries over automatically, but we also don't just drop and forget missed issues).
+
+### Releases
+
+Despite following two-week iterations, our [releases are monthly on the 20th](../../releases.md#releases) and we may sometimes need to order tasks in a way that we can get important projects into the next release.
+
+We also intentionally plan so we can avoid merging significant work less than two days before a release (if a release is on the 20th, our last day to merge is the 18th). Exceptions require explicit approval of both the PM and EM.
+
+### Planning larger projects
+
+Larger (usually multi-iteration) projects coming from our roadmap go through three phases:
+
+1. **Product context and scope agreement** (usually 1-2 weeks)<br>
+   The PM creates a product-level RFC summarizing user desires from feedback and proposes a reasonable timeline to reach a scope agreement (mentioned in the RFC).
+   Engineers, designer (as applicable), PM, and EM discuss the needs in the RFC and the feasibility/complexity of different options.
+   **The goal of this phase is to agree on a scope for the project**, which will generally be a _subset_ of the desires the PM outlined in the RFC based on the implementation complexity, how long it would roughly take to implement, and how much time we want to dedicate to the project.
+   After discussion, the agreed-upon scope is recorded in a section at the bottom of the RFC.
+
+2. **Design and implementation planning** (usually 1-2 weeks)<br>
```

Comment:
I'm not sure about specifying the timelines here. @felixfbecker what's the goal of it?

Reply:
To give a general idea of the timescale of these phases and the whole cycle overall. These can vary per project of course, that's why I wrote "usually".

I considered instead documenting that the timeline should be set explicitly at the start of each phase, but I think it helps to have a rough expectation of timelines before-hand. E.g. Joel shouldn't expect that he can put up a product RFC and then go through the review and design process in a matter of days (not saying he would!), and engineers should have a general understanding of if we're in week 2 of the design phase and the design is ready for review now, they're expected to not put the review at the end of my TODO list for 2 weeks because that would blow the project timeline out of proportion.

While it is good to have a conversation about timelines for every project, having a rough expectation means we don't need to re-negotiate every time whether we're talking about days or months.

It also gives a rough idea of which size project should be to use this process, and at which size they should be broken up into multiple projects. In a way that also helps when selecting the project scope in the first phase (pushing back on a scope that could not be designed or implemented in this order of magnitude of time).

Do you have a concern here? (with documenting a rough timeline at all, or with the specific numbers?)

Reply:
ok, makes sense. thank you